core/variant/variant_op.cpp

```diff
@@ -318,6 +318,7 @@ class OperatorEvaluatorBitAnd {
 		r_valid = true;
 	}
 	static void validated_evaluate(const Variant *left, const Variant *right, Variant *r_ret) {
+		VariantTypeChanger<R>::change(r_ret);
```

Comment:
It would be good to add a comment to explain why this was added, as this isn't 100% obvious at a glance.

Reply:
I modified the comment to hopefully make it more clear.

Reply:
I still think it'd be better to have a code comment, as not everyone will jump to `git blame` instinctively if they reach a line of code they don't understand.

Reply:
I don't necessarily disagree and I did just update the description to make this next point clear, but this line is present for every operator and isn't unique to bit-and. I'd be willing to add a comment to make the importance of this line more clear, but I don't see where it would make sense to add unless copied in-place for every operator.

Reply:
Alright, makes sense then.